automata/fa/dfa.py

```diff
@@ -18,6 +18,62 @@ class OriginEnum(IntEnum):
     OTHER = 1
 
 
+class PartitionRefinement:
```

Comment:
@eliotwrobson Is this a class that can be leveraged by other primitives in this library (e.g. NFA)? Because if so, I think it would be better to move to a separate file.

Reply:
It's probably worth moving to its own file anyway (since this class can be used independently of the DFA). What subheading should this be in? Just under `automata.base`?

I've actually never seen this data structure used before outside of this specific algorithm. Still probably should go in its own file, but interesting that it doesn't come up very often.

Reply:
@eliotwrobson If we go that route, I would suggest the module path be `automata.base.partition_refinement`.

BUT, I was also considering if a `automta.base.utils` module would be better. If we go with this approach, we could also move `OriginEnum` there so that also becomes reusable. Thoughts?

Reply:
I think moving the enum there is a bit overkill. It's very short and only really usable in the equality algorithm. I actually considered making the definition local to that function.

Reply:
@eliotwrobson Yeah, I can see that. But regardless, I think I'd still like to move `PartitionRefinement` under a `automata.base.utils` module. Because it's really just a utility class, and I don't want it to be seen as having the same significance the other library primitives like `.fa.dfa` or `.fa.nfa` or even `.base.automaton`. I also want to create a future-proof place to put ancillary data structures like these.

Reply:
Sounds good! I'll go ahead and move it when I'm back at my desk.

Reply:
@caleb531 class has been moved!